swagger-api/swagger-codegen
# [PHP] Better PSR2 compatibility

### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run`./bin/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

These changes fix the emitted PHP compatibility with the declared [PHP-FIG's PSR2 coding style](http://www.php-fig.org/psr/psr-2/). This can be verified by using the standard tool for the task in the PHP ecosystem, [PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer), for which the configuration has also been included with the `.php_cs` file.

To verify compatibility, run as follows:

    # run swagger-codegen as usual

    # install dev dependencies, which now includes PHP-CS-Fixer
    composer install

    # run PHP-CS-Fixer which should NOT alter any files at all
    vendor/bin/php-cs-fixer fix --dry-run --diff -vvv

### Notes

 - as the emitted code required PHP 5.4 in `composer.json`, array notation has been switched to new short array syntax introduced in [PHP 5.4](http://php.net/manual/en/language.types.array.php) to better match new best practices, this has also been enforced in `.php_cs`
 - additional custom fixers (see [PHP-CS-Fixer's Usage](https://github.com/FriendsOfPHP/PHP-CS-Fixer#usage)) have been enabled to make output more strict (and thus more predictable)
